projects/ng-components-docs/src/app/examples/components/modals/modals.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, ViewChild } from "@angular/core";
+import { ModalSizeType, ModalPositionType } from "lib/src/modal/modal.config";
+import { ModalComponent as MC } from "lib/src/modal/modal.component";
+
+@Component({
+    selector: "app-modal",
+    templateUrl: "./modals.component.html",
+    styleUrls: ["./modals.component.scss"],
+})
+export class ModalComponent {
+    size: ModalSizeType;
+    center: boolean;
+    position: ModalPositionType;
+    fullscreen: boolean;
+    backdropDismiss: boolean = true;
+    @ViewChild(MC) modalChild: MC;
+
+    openModal(): void {
+        this.modalChild.open();
+    }
+
+    openSizeModal(size: ModalSizeType): void {
+        this.size = size;
+        this.openModal();
+    }
+
+    openWithoutBackdrop(): void {
+        this.backdropDismiss = false;
+        this.openModal();
+    }
+
+    openCenteredModal(): void {
+        this.center = true;
+        this.openModal();
+    }
+
+    openFullscreen(): void {
+        this.fullscreen = true;
+        this.openModal();
+    }
+
+    openAside(position: ModalPositionType) {
+        this.position = position;
+        this.openModal();
+    }
+
+    closeModal(): void {
+        this.modalChild.close();
+        setTimeout(() => {
+            this.resetModal();
+        }, 500);
+    }
+
+    resetModal(): void {
+        this.size = null;
+        this.center = null;
+        this.fullscreen = null;
+        this.position = null;
```

Comment:
I think you can simplify this block like this
```suggestion
        this.size = this.center = this.fullscreen = this.position = null;
```

Reply:
and since all of them are doing the same thing `this.openModal()` except for few options, maybe you can merge it into one method with options
```typescript
openModal(options: { size, backdropDismiss, center, fullscreen, position }): void {
}
```